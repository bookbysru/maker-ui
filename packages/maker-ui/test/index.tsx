// import React from 'react'
// import renderer from 'react-test-renderer'
// // import { cleanup } from '@testing-library/react'
// import { matchers } from 'jest-emotion'
// import {
//   Layout,
//   Topbar,
//   Header,
//   Navbar,
//   MobileMenu,
//   Content,
//   Main,
//   SideNav,
//   Sidebar,
//   Footer,
//   Section,
//   Box,
// } from '../src/components'

// expect.extend(matchers)
// // afterEach(cleanup)

// const renderJSON = e => renderer.create(e).toJSON()

// // Remove skiplinks for testing individual components

// const Provider = ({ children, ...props }) => (
//   <Layout options={{ a11y: { skiplinks: false } }} {...props}>
//     {children}
//   </Layout>
// )

// const theme = {
//   colors: {
//     bg_header: '#000',
//   },
//   fontSizes: [8, 10, 14, 18, 24, 32, 48, 64],
//   header: {
//     pt: '20px',
//   },
//   headerTest: {
//     pt: '30px',
//   },
// }

// const options = {
//   layout: 'sidebar-content',
//   header: {
//     sticky: false,
//   },
//   a11y: {
//     skiplinks: false,
//   },
// }

// describe('Layout', () => {
//   test('renders', () => {
//     const json = renderJSON(<Layout>Hi</Layout>)
//     expect(json).toMatchSnapshot()
//   })

//   test('renders without skiplinks', () => {
//     const json = renderJSON(<Layout options={options}>Hello</Layout>)
//     expect(json).toMatchSnapshot()
//   })

//   test('user theme overrides default theme', () => {
//     const json = renderJSON(
//       <Layout options={options} theme={theme}>
//         <Box sx={{ fontSize: 1 }} />
//       </Layout>
//     )
//     expect(json).toHaveStyleRule('font-size', '10px')
//   })

//   test('user options override default options', () => {
//     const json = renderJSON(
//       <Layout options={options}>
//         <Content />
//       </Layout>
//     )
//     expect(json).toHaveStyleRule('display', 'grid')
//   })
// })

// // Uses header component to test integration with Theme UI components

// describe('Header', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Header>Test</Header>
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })

//   test('renders with sx prop', () => {
//     const json = renderJSON(
//       <Provider>
//         <Header sx={{ mb: '10px' }} />
//       </Provider>
//     )
//     expect(json).toHaveStyleRule('margin-bottom', '10px')
//   })

//   test('renders with default variant namespace', () => {
//     const json = renderJSON(
//       <Provider theme={theme}>
//         <Header>Test</Header>
//       </Provider>
//     )
//     expect(json).toHaveStyleRule('padding-top', '20px')
//   })

//   test('renders with custom variant', () => {
//     const json = renderJSON(
//       <Provider theme={theme}>
//         <Header variant="headerTest">Test</Header>
//       </Provider>
//     )
//     expect(json).toHaveStyleRule('padding-top', '30px')
//   })

//   test('renders with prop override', () => {
//     const json = renderJSON(
//       <Layout options={options}>
//         <Header sticky>Test</Header>
//       </Layout>
//     )
//     expect(json).toHaveStyleRule('position', 'sticky')
//   })
// })

// describe('Navbar', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Navbar />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })

//   test('renders with type prop override', () => {
//     const json = renderJSON(
//       <Provider>
//         <Navbar type="reverse" />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('Topbar', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Topbar />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('MobileMenu', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Header>
//           <MobileMenu />
//         </Header>
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('Content', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Content />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('Main', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Content>
//           <Main>Test</Main>
//         </Content>
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })
// describe('Sidebar', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Content>
//           <Sidebar />
//           <Main />
//         </Content>
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })
// describe('SideNav', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <SideNav />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('Footer', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Footer />
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })

// describe('Section', () => {
//   test('renders', () => {
//     const json = renderJSON(
//       <Provider>
//         <Section>Test</Section>
//       </Provider>
//     )
//     expect(json).toMatchSnapshot()
//   })
// })
