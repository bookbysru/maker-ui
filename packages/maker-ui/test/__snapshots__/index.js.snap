// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
  min-height: 80vh;
}

<div
  className="emotion-0"
  id="site-inner"
/>
`;

exports[`Footer renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_footer,#fff);
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<footer
  className="emotion-1"
  id="footer"
  role="contentinfo"
>
  <div
    className="emotion-0"
  />
</footer>
`;

exports[`Header renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 100;
  position: relative;
  background-color: var(--theme-ui-colors-bg_header,#000);
}

<header
  className="emotion-0"
  id="site-header"
  role="banner"
>
  Test
</header>
`;

exports[`Layout renders 1`] = `
Array [
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#000);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

<ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>,
  "Hi",
]
`;

exports[`Layout renders without skiplinks 1`] = `"Hello"`;

exports[`Main renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
  min-height: 80vh;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-1"
  id="site-inner"
>
  <main
    className="emotion-0"
    id="content"
    role="main"
  >
    Test
  </main>
</div>
`;

exports[`MobileMenu renders 1`] = `
.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 100;
  position: relative;
  background-color: var(--theme-ui-colors-bg_header,#000);
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.15);
  z-index: 100;
  visibility: hidden;
  opacity: 0;
  will-change: opacity;
  -webkit-transition: all ease .4s;
  transition: all ease .4s;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  background-color: var(--theme-ui-colors-bg_mobileMenu,rgba(0,0,0,0.9));
  top: 0;
  bottom: 0;
  z-index: 100;
  will-change: transform opacity;
  -webkit-transition: all ease .3s;
  transition: all ease .3s;
  opacity: 1;
  visibility: hidden;
  left: 0;
  width: 60vw;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

<header
  className="emotion-3"
  id="site-header"
  role="banner"
>
  <div
    className="menu-overlay emotion-0"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="emotion-2"
    id="mobile-menu"
  >
    <ul
      className="accordion-menu emotion-1"
    />
  </div>
</header>
`;

exports[`Navbar renders 1`] = `
.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  margin: 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1460px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: initial;
  -ms-flex: initial;
  flex: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
}

@media screen and (min-width:750px) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  margin: 0 auto;
  border: none;
  background: none;
}

@media screen and (min-width:750px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-4 svg {
  margin: 0 auto;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  margin: 0;
  height: 27px;
}

<div
  className="emotion-7"
>
  <a
    aria-label="Home page"
    className="emotion-0"
    href="/"
    id="site-logo"
  >
    Logo
  </a>
  <div
    className="emotion-6"
  >
    <nav
      className="nav-primary emotion-2"
    >
      <ul
        className="menu-primary emotion-1"
      />
    </nav>
    <div
      className="emotion-5"
    >
      <button
        aria-expanded="false"
        aria-label="Toggle Menu"
        className="menu-toggle emotion-4"
        onClick={[Function]}
        title="Menu"
      >
        <svg
          className="emotion-3"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Navbar renders with type prop override 1`] = `
.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
}

@media screen and (min-width:750px) {
  .emotion-1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  margin: 0 auto;
  border: none;
  background: none;
}

@media screen and (min-width:750px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-3 svg {
  margin: 0 auto;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  margin: 0;
  height: 27px;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  margin: 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 1460px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 .col-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25%;
}

@media screen and (min-width:750px) {
  .emotion-8 .col-1 {
    width: 33%;
  }
}

.emotion-8 .col-1 button {
  margin: 0;
}

.emotion-8 .col-2 {
  width: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (min-width:750px) {
  .emotion-8 .col-2 {
    width: 34%;
  }
}

.emotion-8 .col-3 {
  width: 25%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media screen and (min-width:750px) {
  .emotion-8 .col-3 {
    width: 33%;
  }
}

<div
  className="emotion-8"
>
  <div
    className="col-1 emotion-4"
  >
    <nav
      className="nav-primary emotion-1"
    >
      <ul
        className="menu-primary emotion-0"
      />
    </nav>
    <button
      aria-expanded="false"
      aria-label="Toggle Menu"
      className="menu-toggle emotion-3"
      onClick={[Function]}
      title="Menu"
    >
      <svg
        className="emotion-2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
  <div
    className="col-2 emotion-4"
  >
    <a
      aria-label="Home page"
      className="emotion-4"
      href="/"
      id="site-logo"
    >
      Logo
    </a>
  </div>
  <div
    className="col-3 emotion-4"
  />
</div>
`;

exports[`Section renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
}

<section
  className="emotion-1"
>
  <div
    className="emotion-0"
  >
    Test
  </div>
</section>
`;

exports[`SideNav renders 1`] = `
Array [
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.15);
  z-index: 100;
  visibility: hidden;
  opacity: 0;
  will-change: opacity;
  -webkit-transition: all ease .4s;
  transition: all ease .4s;
}

@media screen and (min-width:750px) {
  .emotion-0 {
    visibility: hidden;
    opacity: 0;
  }
}

<div
    className="menu-overlay emotion-0"
    onClick={[Function]}
    role="presentation"
  />,
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_sideNav,#333);
  position: fixed;
  top: 0;
  bottom: 0;
  z-index: 100;
  width: 300px;
  will-change: transform;
  -webkit-transform: translateX(300px);
  -ms-transform: translateX(300px);
  transform: translateX(300px);
  -webkit-transition: -webkit-transform ease .3s;
  -webkit-transition: transform ease .3s;
  transition: transform ease .3s;
}

@media screen and (min-width:750px) {
  .emotion-1 {
    position: relative;
    z-index: 0;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<div
    className="emotion-1"
    id="side-nav"
    role="navigation"
  >
    <ul
      className="accordion-menu emotion-0"
    />
  </div>,
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  display: inline-block;
  bottom: 30px;
  z-index: 100;
}

@media screen and (min-width:750px) {
  .emotion-0 {
    display: none;
  }
}

<button
    aria-label="Toggle side navigation"
    className="emotion-0"
    id="toggle-sidenav"
    onClick={[Function]}
    title="Toggle SideNav"
  >
    Toggle
  </button>,
]
`;

exports[`Sidebar renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
  min-height: 80vh;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-2"
  id="site-inner"
>
  <aside
    className="emotion-0"
    id="primary-sidebar"
    role="complementary"
  />
  <main
    className="emotion-1"
    id="content"
    role="main"
  />
</div>
`;

exports[`Topbar renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_topbar,#355cac);
  display: block;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 1260px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<aside
  className="emotion-1"
  role="complementary"
>
  <div
    className="emotion-0"
  />
</aside>
`;
